README.md

```diff
@@ -2,4 +2,134 @@
+* Integrated mode will closely integrate with the Scripts directory and bind-mount it as well as the portage-stable and coreos-overlay gitmodules into the container. Integrated mode uses wrapper scripts to interact with the SDK container. This is the recommended way for developing patches for Flatcar.
+
+## Standalone mode
+
+In standalone mode, the SDK is just another Docker container. Interaction with the container happens via use of `docker` directly. Use for experimenting and for throw-away work only, otherwise please use integrated mode (see below).
+
+* Check the list of available versions and pick a version to use. The SDK Major versions correspond to Flatcar Major release versions.
+  List of images: `https://github.com/orgs/flatcar-linux/packages/container/package/flatcar-sdk-all`
+  For the purpose of this example we'll use version `3033.0.0`.
+* Fetch the container image: `docker pull ghcr.io/flatcar-linux/flatcar-sdk-all:3033.0.0`
+* Start the image in interactive (tty) mode: `docker run -ti ghcr.io/flatcar-linux/flatcar-sdk-all:3033.0.0`
+  You are now inside the SDK container (the hostname will likely differ):
+  `sdk@f236fda982a4 ~/trunk/src/scripts $`
+* Initialise the SDK in self-contained mode. This needs to be done once per container and will check out the scripts, coreos-overlay, and portage-stable repositories into the container.
+  `sdk@f236fda982a4 ../sdk_init_selfcontained.sh`
+
+You can now work with the SDK container.
+
+### Privileged mode when building images
+
+In order to build OS images (via `./build_image` and `./image_to_vm`) the SDK tooling requires privileged access to `/dev`.
+This is necessary because the SDK currently employs loop devices to create and to partition OS images.
+
+To start a container in privileged mode with `/dev` available use:
+* `docker run -ti  --privileged -v /dev:/dev ghcr.io/flatcar-linux/flatcar-sdk-all:3033.0.0`
+
+## Integrated mode
+
+This is the preferred mode of working with the SDK.
+Interaction with the container happens via wrapper scripts from the Scripts repository.
+Both the host's scripts repo as well as its submodules (portage-stable and coreos-overlay) are made available in the container, allowing for work on these repos directly.
+The wrapper scripts will re-use existing containers instead of creating new ones to preserve your work in the container, enabling consistency.
+
+To clone the scripts repo and pick a version:
+* Clone the scripts repo: `git clone https://github.com/flatcar-linux/scripts.git`
+  * Make sure to initialise and fetch git submodules - Flatcar's ebuilds are in 2 separate repositories, connected to `scripts` via submodules.
+  * `git submodule init; git submodule update`
+* Optionally, check out a release tag to base your work on
+  * list releases (e.g. all Alpha releases): `git tag -l alpha-*`
+  * check out the release version, e.g. `3033.0.0`: `git checkout 3033.0.0`
```

Comment:
I think it should more or less be:
- Clone the scripts repo
- Optionally, check our a release tag
- Initialize submodules

Initialization of submodules should happen after picking a tag as a base, since it could have different submodule checkouts.